### PR TITLE
Update py-pint versions

### PR DESCRIPTION
**Pull request by lpritch-lanl**

### CODE REVIEW

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Missing the following dependencies:
```
python_requires = >=3.6
install_requires = setuptools
setup_requires = setuptools; setuptools_scm
```
Also, is there a reason to switch from PyPI to GitHub?

**Comment by lpritch-lanl** [comment]
> Missing the following dependencies:
> 
> ```
> python_requires = >=3.6

I believe I have fixed this.

> install_requires = setuptools
> setup_requires = setuptools; setuptools_scm
> ```
> 

I don't know how to resolve these. I have a depends_on('py-setuptools') statement in the package.py file, but could not locate a setuptools-scm package.

> Also, is there a reason to switch from PyPI to GitHub?

I originally had issues downloading from PyPI that I resolved. Switched it back to PyPI.

**Comment by adamjstewart** [comment]
> I don't know how to resolve these. I have a depends_on('py-setuptools') statement in the package.py file, but could not locate a setuptools-scm package.

There is a `py-setuptools-scm` package, you just need to add a build dependency on that.

File: var/spack/repos/builtin/packages/py-pint/package.py [hunk 1]
```diff
@@ -12,9 +12,19 @@ class PyPint(PythonPackage):
...
-    version('0.8.1', sha256='afcf31443a478c32bbac4b00337ee9026a13d0e2ac83d30c79151462513bb0d4')
+    version(
+        '0.11', sha256='308f1070500e102f83b6adfca6db53debfce2ffc5d3cbe3f6c367da359b5cf4d')
+    version('0.10.1', sha256='d739c364b8326fe3d70773d5720fa8b005ea6158695cad042677a588480c86e6')
+    version(
+        '0.10', sha256='38a4d6e242b8bab693cd83a5f5ade3d816463b498658e7ab14ce64c4d458c88b')
+    version(
+        '0.9', sha256='32d8a9a9d63f4f81194c0014b3b742679dce81a26d45127d9810a68a561fe4e2')
+    version(
+        '0.8.1', sha256='afcf31443a478c32bbac4b00337ee9026a13d0e2ac83d30c79151462513bb0d4')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pint/package.py]: These can be on a single line, we exempt checksums from line length requirements.

Comment (lpritch-lanl) [reply var/spack/repos/builtin/packages/py-pint/package.py]: Placed them back on the same line as version

File: var/spack/repos/builtin/packages/py-pint/package.py [hunk 2]
```diff
@@ -12,9 +12,16 @@ class PyPint(PythonPackage):
...
 
+    version('0.11', sha256='308f1070500e102f83b6adfca6db53debfce2ffc5d3cbe3f6c367da359b5cf4d')
+    version('0.10.1', sha256='d739c364b8326fe3d70773d5720fa8b005ea6158695cad042677a588480c86e6')
+    version('0.10', sha256='38a4d6e242b8bab693cd83a5f5ade3d816463b498658e7ab14ce64c4d458c88b')
+    version('0.9', sha256='32d8a9a9d63f4f81194c0014b3b742679dce81a26d45127d9810a68a561fe4e2')
     version('0.8.1', sha256='afcf31443a478c32bbac4b00337ee9026a13d0e2ac83d30c79151462513bb0d4')
 
-    depends_on('py-setuptools', type='build')
+    depends_on('python@3.6:', type=('build', 'run'), when='@0.10:')
+    depends_on('python', type=('build', 'run'), when='@0.8.1,0.9')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pint/package.py]: This line isn't necessary, all `PythonPackage`s depend on Python

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**